tests/core/asyncio/test_broadcast_config.py

```diff
@@ -1,61 +1,70 @@
+import asyncio
+
 import pytest
 
-from helpers import DummyRequestPair, DummyResponse, Tracker
-from lahja import BroadcastConfig
+from lahja import BaseEvent, BroadcastConfig
+
+
+class BroadcastEvent(BaseEvent):
+    pass
 
 
 @pytest.mark.asyncio
-async def test_broadcasts_to_all_endpoints(triplet_of_endpoints):
-    endpoint1, endpoint2, endpoint3 = triplet_of_endpoints
+async def test_broadcasts_to_all_endpoints(server_with_two_clients):
+    server, client_a, client_b = server_with_two_clients
 
-    tracker = Tracker()
+    return_queue = asyncio.Queue()
 
-    endpoint1.subscribe(
-        DummyRequestPair, tracker.track_and_broadcast_dummy(1, endpoint1)
-    )
+    client_a.subscribe(BroadcastEvent, return_queue.put_nowait)
+    client_b.subscribe(BroadcastEvent, return_queue.put_nowait)
 
-    endpoint2.subscribe(
-        DummyRequestPair, tracker.track_and_broadcast_dummy(2, endpoint2)
-    )
+    await server.wait_until_all_remotes_subscribed_to(BroadcastEvent)
+    await server.broadcast(BroadcastEvent())
 
-    await endpoint3.wait_until_all_remotes_subscribed_to(DummyRequestPair)
+    result_a = await asyncio.wait_for(return_queue.get(), timeout=0.1)
+    result_b = await asyncio.wait_for(return_queue.get(), timeout=0.1)
 
-    item = DummyRequestPair()
-    response = await endpoint3.request(item)
-    assert isinstance(response, DummyResponse)
-    assert tracker.exists(1)
-    assert tracker.exists(2)
-    # Ensure the registration was cleaned up
-    assert item._id not in endpoint3._futures
+    assert isinstance(result_a, BroadcastEvent)
+    assert isinstance(result_b, BroadcastEvent)
 
-    endpoint1.stop()
-    endpoint2.stop()
-    endpoint3.stop()
 
+class TailEvent(BaseEvent):
+    pass
 
-@pytest.mark.asyncio
-async def test_broadcasts_to_specific_endpoint(triplet_of_endpoints):
 
-    endpoint1, endpoint2, endpoint3 = triplet_of_endpoints
+@pytest.mark.asyncio
+async def test_broadcasts_to_specific_endpoint(server_with_two_clients):
```

Comment:
I like this, it's an improvement over how the test was! However, there are now no tests for `Endpoint.request(..., BC(filter_endpoint=...))`, that sounds worth testing, maybe there could be an additional test after this one which tests the same thing but while making a request?